bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java

```diff
@@ -453,31 +461,48 @@ String getSQL() {
 	 *
 	 * @throws TskCoreException
 	 */
-	private long addEventDescription(long dataSourceObjId, long fileObjId, Long artifactID,
+	private Long addEventDescription(long dataSourceObjId, long fileObjId, Long artifactID,
 			String fullDescription, String medDescription, String shortDescription,
 			boolean hasHashHits, boolean tagged, CaseDbConnection connection) throws TskCoreException {
-		String insertDescriptionSql
-				= "INSERT INTO tsk_event_descriptions ( "
+		String tableValuesClause
+				= "tsk_event_descriptions ( "
 				+ "data_source_obj_id, content_obj_id, artifact_id,  "
 				+ " full_description, med_description, short_description, "
 				+ " hash_hit, tagged "
-				+ " ) VALUES ("
-				+ dataSourceObjId + ","
-				+ fileObjId + ","
-				+ Objects.toString(artifactID, "NULL") + ","
-				+ quotePreservingNull(fullDescription) + ","
-				+ quotePreservingNull(medDescription) + ","
-				+ quotePreservingNull(shortDescription) + ", "
-				+ booleanToInt(hasHashHits) + ","
-				+ booleanToInt(tagged)
-				+ " )";
+				+ " ) VALUES "
+				+ "(?, ?, ?, ?, ?, ?, ?, ?)";
+
+		String insertDescriptionSql = getSqlIgnoreConflict(tableValuesClause);
 
 		caseDB.acquireSingleUserCaseWriteLock();
-		try (Statement insertDescriptionStmt = connection.createStatement()) {
-			connection.executeUpdate(insertDescriptionStmt, insertDescriptionSql, PreparedStatement.RETURN_GENERATED_KEYS);
+		try (PreparedStatement insertDescriptionStmt = connection.prepareStatement(insertDescriptionSql, PreparedStatement.RETURN_GENERATED_KEYS)) {
+			insertDescriptionStmt.clearParameters();
+			insertDescriptionStmt.setLong(1, dataSourceObjId);
+			insertDescriptionStmt.setLong(2, fileObjId);
+
+			if (artifactID == null) {
+				insertDescriptionStmt.setNull(3, Types.INTEGER);
+			} else {
+				insertDescriptionStmt.setLong(3, artifactID);
+			}
+
+			insertDescriptionStmt.setString(4, fullDescription);
+			insertDescriptionStmt.setString(5, medDescription);
+			insertDescriptionStmt.setString(6, shortDescription);
+			insertDescriptionStmt.setInt(7, booleanToInt(hasHashHits));
+			insertDescriptionStmt.setInt(8, booleanToInt(tagged));
+			int row = insertDescriptionStmt.executeUpdate();
+			// if no inserted rows, return null.
```

Comment:
I think we need a comment here (and at a few other points where we are dealing with null event description IDs) to explain that we got no rows because the description is already in the database and therefore making the event would be creating a duplicate would be helpful.